editor/src/components/canvas/commands/push-intended-bounds-and-update-targets-command.ts

```diff
@@ -42,102 +45,270 @@ import {
+      editor,
+      groupTargets,
+      commandRanBecauseOfQueuedTrueUp,
+      commandLifecycle,
+    )
+  editorStatePatches.push(...groupsEditorStatePatches)
+
+  // Hugging elements
+  const huggingTargets = command.value.filter(isPushIntendedBoundsTargetHuggingElement)
+  const { updatedEditor: editorAfterHuggingElements } =
+    runPushIntendedBoundsAndUpdateTargetsHuggingElement(editorAfterGroups, huggingTargets)
+
+  // TODO this is the worst editor patch in history, this should be much more fine grained, only patching the elements that changed
+  editorStatePatches.push({
+    projectContents: {
+      $set: editorAfterHuggingElements.projectContents,
+    },
+    toasts: {
+      $set: editorAfterHuggingElements.toasts,
+    },
+  })
+
+  return {
+    editorStatePatches: editorStatePatches,
+    commandDescription: `Set Intended Bounds for ${command.value
+      .map((c) => EP.toString(c.target))
+      .join(', ')}`,
+  }
+}
+
+function runPushIntendedBoundsAndUpdateTargetsGroup(
+  editor: EditorState,
+  targets: Array<PushIntendedBoundsTargetGroup>,
+  commandRanBecauseOfQueuedTrueUp: boolean,
+  commandLifecycle: InteractionLifecycle,
+): {
+  updatedEditor: EditorState
+  editorStatePatches: Array<EditorStatePatch>
+} {
   const { updatedEditor: editorAfterResizingGroupChildren, resizedGroupChildren } =
-    getUpdateResizedGroupChildrenCommands(editor, command)
+    getUpdateResizedGroupChildrenCommands(editor, targets)
 
   const {
     updatedEditor: editorAfterResizingAncestors,
     intendedBounds: resizeAncestorsIntendedBounds,
   } = getResizeAncestorGroupsCommands(
     editorAfterResizingGroupChildren,
-    command,
+    targets,
     commandRanBecauseOfQueuedTrueUp
       ? 'do-not-create-if-doesnt-exist'
       : // TODO this can be removed in a future PR, but for now matching the previous behavior
         'create-if-not-existing',
   )
 
-  // TODO this is the worst editor patch in history, this should be much more fine grained, only patching the elements that changed
-  const editorPatch = {
-    projectContents: { $set: editorAfterResizingAncestors.projectContents },
-    toasts: { $set: editorAfterResizingAncestors.toasts },
-  }
-
-  const intendedBoundsPatch =
-    commandLifecycle === 'mid-interaction'
-      ? pushIntendedBoundsPatch([...command.value, ...resizeAncestorsIntendedBounds])
-      : []
-
-  const queueFollowUpGroupTrueUpPatch: Array<EditorStatePatch> =
-    commandLifecycle === 'end-interaction' && !commandRanBecauseOfQueuedTrueUp
-      ? [
-          {
-            trueUpGroupsForElementAfterDomWalkerRuns: {
-              $set: resizedGroupChildren.map((c) => trueUpElementChanged(c)),
-            },
+  let editorStatePatches: Array<EditorStatePatch> = []
+  if (commandLifecycle === 'mid-interaction') {
+    editorStatePatches.push({
+      canvas: {
+        controls: {
+          strategyIntendedBounds: {
+            $push: [...targets, ...resizeAncestorsIntendedBounds],
```

Comment:
I wouldn't push the intended bounds here in the runPushIntendedBoundsAndUpdateTargetsGroup